### ISSUE TITLE
const optimization on strings #1

### ISSUE THREAD

**Issue by chuacw** [IndySockets/Indy#195]
procedure TIdCmdTCPClient.DoReplyUnknownCommand(AContext: TIdContext; ALine: string) in IdCmdTCPClient 

to change to

procedure TIdCmdTCPClient.DoReplyUnknownCommand(AContext: TIdContext; const ALine: string).

**Comment by rlebeau** [comment]
Why?

**Comment by rlebeau** [comment]
Why?

**Comment by chuacw** [comment]
Improve performance of Indy, so that the invisible calls to UAddStrRef and UStrClr, in addition to a try finally block are removed.

**Comment by rlebeau** [comment]
Most internet protocols have textual elements to them, so Indy is very heavily string-oriented and has a lot of string processing functionality.  What you are asking for is not just a few select optimizations, but really a complete review of the entire codebase. We don't have the manpower for that level of effort.

**Comment by WhiteWind** [comment]
But this is not a complete review if codebase. It's just about adding const
specifier to string parameters. And it doesn't break interface - it only
changes the implementation of method by removing implicit try/finally block
and refcounting on const pointer

ср, 7 марта 2018 г., 4:34 Remy Lebeau <notifications@github.com>:

**Comment by rlebeau** [comment]
Yes, it would need to be a complete review. Like I said, Indy is pretty heavily string-oriented. Do you realize how many string parameters Indy actually has?  A LOT.  And yes, adding 'const' to string parameters is a DCU-breaking interface change.  I try to keep interface changes to a minimum except in big releases.  This would be too much of a sweeping change for the current release.  However, there is a maintenance release currently in progress that already has some big sweeping changes in it, so I *might* consider this issue for that release, or it may have to wait for the next major version instead.
